### PR TITLE
chore(client): move the client to its own subfolder 

### PR DESCRIPTION
**Pull request by daniellacosse**
I was able to successfully move the client into its own subfolder:

Screenshot 2024-02-12 at 1 49 21 PM

TODO:

- [x] should i move build into the root? or try to mess with `ROOT_DIR`?
- [x] basic documentation
- [x] move third_party into the root (need help from @jyyi1)
- [ ] unbreak it... again
  - [x] electron
  - [ ] windows (it can't find the env.nsh)
  - [x] android working in the CI (dialer.Dial undefined??)
  - [x] maccatalyst working in the CI

### CODE REVIEW

File: client/third_party/jsign/.gitignore
Comment (fortuna) [inline client/third_party/jsign/.gitignore:1]: third_party must stay in the root:
- http://go/thirdparty/non-google3#WhereInThirdParty
- http://go/thirdparty/where#separate-out-dependencies

File: README.md [hunk 1]
```diff
@@ -1,93 +1,3 @@
...
-export SENTRY_DSN=[Sentry development API key]
-[platform-specific build command]
-```
-
-Release builds on CI are configured with a production Sentry API key.
-
-## Support
-
-For support and to contact us, see: https://support.getoutline.org.
+TODO
```
Comment (fortuna) [inline README.md:3]: This is a must have. At a minimum we should link to the client and the manager

File: README.md [hunk 2]
```diff
@@ -1,93 +1,3 @@
-# Outline Client
+# Outline Apps
 
-![Build and Test](https://github.com/Jigsaw-Code/outline-client/actions/workflows/build_and_test_debug.yml/badge.svg?branch=master) [![Mattermost](https://badgen.net/badge/Mattermost/Outline%20Community/blue)](https://community.internetfreedomfestival.org/community/channels/outline-community) [![Reddit](https://badgen.net/badge/Reddit/r%2Foutlinevpn/orange)](https://www.reddit.com/r/outlinevpn/)
```
Comment (fortuna) [inline README.md:3]: This should probably stay here?

File: client/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
+# Outline Client
+
+![Build and Test](https://github.com/Jigsaw-Code/outline-apps/actions/workflows/build_and_test_debug.yml/badge.svg?branch=master) [![Mattermost](https://badgen.net/badge/Mattermost/Outline%20Community/blue)](https://community.internetfreedomfestival.org/community/channels/outline-community) [![Reddit](https://badgen.net/badge/Reddit/r%2Foutlinevpn/orange)](https://www.reddit.com/r/outlinevpn/)
```
Comment (fortuna) [inline client/README.md:3]: @https://github.com/Jigsaw-Code/outline-apps/actions/workflows/build_and_test_debug.yml/badge.svg?branch=master is not found. It was broken in the previous PR

File: tools/OutlineService/OutlineService/bin/Release/OutlineService.exe.config
Comment (fortuna) [inline tools/OutlineService/OutlineService/bin/Release/OutlineService.exe.config:1]: Why is this being deleted?

**Review by fortuna — CHANGES_REQUESTED** [review]
A number of tests are still failing, so I think there's still stuff to figure out.

File: client/third_party/jsign/.gitignore
Comment (jyyi1) [reply client/third_party/jsign/.gitignore:1]: I think Vini means we don't touch or move anything in `third_party`, and make sure it's in the root path. There should be only one `third_party` folder in the repository root.

**Review by jyyi1 — COMMENTED** [review]
For the Windows build failure, we just need to specify the `projectDir` when [calling electron-builder](https://github.com/Jigsaw-Code/outline-apps/blob/f3e10d6a5a9012bf1ceba165a04d53e39cc17915/src/electron/build.action.mjs#L64-L82) (currently it's `D:\a\outline-apps\outline-apps`, but it should be `D:\a\outline-apps\outline-apps\client` because [`env.nsh` will be generated there](https://github.com/Jigsaw-Code/outline-apps/blob/f3e10d6a5a9012bf1ceba165a04d53e39cc17915/src/electron/build_main.action.mjs#L60)), similar to [what we did in manager](https://github.com/Jigsaw-Code/outline-apps/blob/f3e10d6a5a9012bf1ceba165a04d53e39cc17915/server_manager/electron_app/package.action.sh#L26).

**Comment by jyyi1** [comment]
> @jyyi1 I've updated the SDK - didn't you say some naming changed? What should I fix

Basically anytime you see an error: `dialer.Dial undefined (type transport.StreamDialer has no field or method Dial)`, just rename the `Dial` in the source code to `DialStream` (if the error message mentions `StreamDialer`) or `DialPacket` (if the error message mentions `PacketDialer`).

File: server_manager/electron_app/build.action.sh [hunk 4]
```diff
@@ -50,7 +50,7 @@ if [[ -z "${WEBPACK_MODE:-}" ]]; then
 fi
 
 # Build the Web App.
-node src/build/run_action.mjs server_manager/web_app/build
+node client/src/build/run_action.mjs server_manager/web_app/build
```
Comment (fortuna) [inline server_manager/electron_app/build.action.sh:53]: Why is the Manager referring to the client?

**Review by fortuna — CHANGES_REQUESTED** [review]

**Comment by fortuna** [comment]
@daniellacosse can we perhaps only migrate Cordova in this PR so we can move forward quickly?

**Pull request closed without merging**